ci/official/any.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+source "${BASH_SOURCE%/*}/utilities/setup.sh"
+
+# Parse options and build targets into arrays, so that shelllint doesn't yell
+# about readability. We can't pipe into 'read -ra' to create an array because
+# piped commands run in subshells, which can't store variables outside of the
+# subshell environment.
+# Ignore grep failures since we're using it for basic filtering
+set +e
+filtered_build_targets=( $(echo "$BUILD_TARGETS" | tr ' ' '\n' | grep . | tee build_targets.txt) )
+nonpip_targets=( $(echo "$TEST_TARGETS" | tr ' ' '\n' | grep -E "^//tensorflow/" | tee nonpip_targets.txt) )
+config=( $(echo "$CONFIG_OPTIONS" ) )
+test_flags=( $(echo "$TEST_FLAGS" ) )
+set -e
+
+[[ "$TFCI_NVIDIA_SMI_ENABLE" == 1 ]] && tfrun nvidia-smi
```

Comment:
I am somewhat conflicted on these commands and if they should just be put in a proper if statement instead of using && syntax which i find a little more difficult to read.  What are your thoughts here?

Reply:
Right, the difference is:

```
[[ "$TFCI_NVIDIA_SMI_ENABLE" == 1 ]] && tfrun nvidia-smi
```
or
```
if [[ "$TFCI_NVIDIA_SMI_ENABLE" == 1 ]]; then
    tfrun nvidia-smi
fi
```

Personally, I like placing simple conditionals in one line, because it feels like a fully-encapsulated statement. However, I also believe these scripts should err towards readability whenever possible, and bash's multiple styles of command chaining are definitely esoteric for the average reader--so I'll change all of the conditionals to explicit `if`s.